shopinvader_quotation/services/quotation.py

```diff
@@ -58,6 +59,9 @@ def _get_base_search_domain(self):
         ]
 
     def _confirm_cart(self, quotation):
+        # TODO: Check when is this method called???
```

Comment:
@sebastienbeau any clue? I think this method is dead

Reply:
I think to dead too
It have been added here : https://github.com/shopinvader/odoo-shopinvader/pull/174/files 
At that moment we where using the payment_gateway module (that have been drop)
So you can remove it.

Reply:
@bguillot ping

Reply:
Hello @sebastienbeau, thanks for your input :)

As this method's not used, then [action_confirm_cart in shopinvader module](https://github.com/shopinvader/odoo-shopinvader/blob/962fb93f486f9550858cbdbe3a2a0b935dc242a2/shopinvader/models/sale.py#L61-L71) isn't used either (apart from tests), and so I don't understand how a cart can be confirmed into a sale through the api.

It's related to my PR in the sense that I'm guessing there also should be a way to convert a "quotation" into a cart/sale, and it should probably be done in a similar way. This was clearly the developer's intention with `_confirm_cart`, even though it's not used.

Reply:
and this is the only ref I found https://github.com/shopinvader/odoo-shopinvader-payment/commit/c7fd687e7bbc526e6416430dfd220f12a6b74e35

still, it looks weird that from the frontend you have no way to confirm a quotation, no?

Reply:
FTR I've talked w/ @PierrickBrun and he told me the reason was that Akretion's customer was only creating request via endpoints and then quotation were handled from the backend.

I would do this:

1. rename the method to something more meaningful, eg: `_convert_to_cart` or `_transform_to_cart` or similar
2. add an endpoint that checks if you already have a cart in the session. If yes, raise an error, if not convert it to cart

What do you think?

Reply:
Agree :+1: